src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { fireEvent, act } from '@testing-library/react-native';
+import SegmentControl from './../SegmentControl';
+import { renderWithTheme } from '../../../_helpers/testing';
+
+describe('<SegmentControl />', () => {
+  describe('variant', () => {
+    it('renders outlined segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="outlined">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+
+    it('renders filled segment control ', () => {
+      const { container } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="filled">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('controlled', () => {
+    it('outlined segment control ', () => {
+      const mockPress = jest.fn();
+
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl value="Option 1" variant="outlined" onChange={mockPress}>
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+      fireEvent.press(option);
+      expect(mockPress).toBeCalledTimes(1);
```

Comment:
Can have more more assertion like `expect(mockPress).toBeCalledWith("Option 2")`, just to confirm if user is expecting the same value in the callback function.